src/udiskslinuxblock.c

```diff
@@ -4196,6 +4196,97 @@ handle_rescan (UDisksBlock           *block,
 
 /* ---------------------------------------------------------------------------------------------------- */
 
+/* runs in thread dedicated to handling method call */
+static gboolean
+handle_restore_encrypted_header (UDisksBlock           *encrypted,
+                                 GDBusMethodInvocation *invocation,
+                                 const gchar           *backup_file,
+                                 GVariant              *options)
+{
+    UDisksObject *object = NULL;
+    UDisksBlock *block;
+    UDisksDaemon *daemon;
+    UDisksState *state = NULL;
+    uid_t caller_uid;
+    GError *error = NULL;
+    UDisksBaseJob *job = NULL;
+
+    object = udisks_daemon_util_dup_object (encrypted, &error);
+    if (object == NULL)
+      {
+        g_dbus_method_invocation_return_gerror (invocation, error);
+        goto out;
+      }
+
+    block = udisks_object_peek_block (object);
+    daemon = udisks_linux_block_object_get_daemon (UDISKS_LINUX_BLOCK_OBJECT (object));
+    state = udisks_daemon_get_state (daemon);
+
+    udisks_linux_block_object_lock_for_cleanup (UDISKS_LINUX_BLOCK_OBJECT (object));
+    udisks_state_check_block (state, udisks_linux_block_object_get_device_number (UDISKS_LINUX_BLOCK_OBJECT (object)));
+
+    /* Fail if the device is not a LUKS device */
```

Comment:
This will fail if the header is completely wiped or corrupted in a way that `blkid` won't recognize it as a LUKS device.